src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/TopologyHelper.groovy

```diff
@@ -207,6 +207,15 @@ class TopologyHelper {
         }
     }
 
+    SwitchTriplet findSwitchTripletWithSharedEpInTheMiddleOfTheChain() {
+        def pairSharedEpAndEp1 = getAllSwitchPairs().neighbouring().random()
+        //shared endpoint should be in the middle of the switches chain to deploy ha-flow without shared path
+        def pairEp1AndEp2 = getAllSwitchPairs().neighbouring().excludePairs([pairSharedEpAndEp1])
+                .includeSwitch(pairSharedEpAndEp1.src).random()
+        return new SwitchTriplet(shared: pairSharedEpAndEp1.src, ep1: pairSharedEpAndEp1.dst,
+                ep2: pairSharedEpAndEp1.src == pairEp1AndEp2.dst ? pairEp1AndEp2.src : pairEp1AndEp2.dst)
```

Comment:
I'm not following the new design too close, but the returned object looks incomplete without having proper `pathsEp1` and `pathsEp2` fields filled. This may lead to unexpected behavior for the 'users' of the returned switchTriplet object that may want to access its paths. This also does not correspond to other methods here that do return complete SwitchTriplet-s (with paths).

Reply:
sure, the requested part has been updated.